blog/_posts/2018-03-29-missing.md

```diff
@@ -0,0 +1,394 @@
+improvements in the Julia compiler and language which make it fast, making it possible
+to allow drop the `DataArray` type and using the standard `Array` type instead[^PDA].
+
+[^PDA]: The `PooledDataArray` type shipped in the same package can be replaced with
+either [`CategoricalArray`](https://github.com/JuliaData/CategoricalArrays.jl) or
+[`PooledArray`](https://github.com/JuliaComputing/PooledArrays.jl) depending on whether
+the data is really categorical or simply contains a small number of distinct values.
+
+This framework is used by
+[version 0.11](https://discourse.julialang.org/t/dataframes-0-11-released/7296/)
+of the [DataFrames](https://github.com/JuliaStats/DataFrames.jl/) package,
+which already works on Julia 0.6, even if performance improvements
+will only become available with Julia 0.7
+
+The new `Missing` type also replaces the `Nullable` type introduced in Julia 0.4,
+which turned out not to be the best choice to represent missing values[^jmw]. `Nullable`
+suffered from several issues:
+
+[^jmw]: In [a 2014 blog post](http://www.johnmyleswhite.com/notebook/2014/11/29/whats-wrong-with-statistics-in-julia/),
+John Myles White advocated the use of `Nullable` due to its much higher performance
+compared with`Union{T,NA}`. This performance gap no longer exists thanks to compiler
+improvements which have been a game-changer for missing values support.
+
+- It was used to represent two very different kinds of missingness, that we
+  sometimes call respectively the "software engineer's null" and the "data
+  scientist's null". The former refers to what `Nullable` is generally used for
+  in most languages, i.e. `null` denotes the absence of a value, and one of the
+  advantages of the `Nullable` type is to force developers to handle explicitly
+  the case when there is no value. The latter refers to statistical missing values,
+  which generally *propagate* silently in specialized languages, which is essential
+  for convenience. It has become increasingly clear that these two uses conflict
+  (even if specialized syntax could have helped mitigating this).
+
+- To ensure type-stability, the `T` type parameter of `Nullable{T}` had to be
+  specified whether a value was present or not. Finding out the appropriate type
+  in the absence of any value (dubbed the "counterfactual return type") has
+  turned out to be problematic in many cases, where code had to rely heavily
+  on explicit calls to type inference, which would better be handled directly
+  by the compiler.
+
+- `Array{Nullable{T}}` objects used a sub-optimal memory layout where `T` values
+  and the associated `Bool` indicator were stored side-by-side, which wastes
+  space due to aligment constraints and is not the most efficient for processing.
+  Therefore, specialized array types like
+  [`NullableArray`](https://github.com/JuliaStats/NullableArrays.jl) had to be used
+  (similar to `DataArray`).
+
+For all these reasons, `Nullable{T}` will
```

Comment:
I think I used the future in the rest of the post to imply that these features are going to be available once the release is out. To avoid the ambiguity here, I've changed the phrasing to "no longer exists".